integration-tests/src/test/groovy/com/okta/sdk/tests/it/ApplicationsIT.groovy

```diff
@@ -517,4 +524,106 @@ class ApplicationsIT extends ITSupport {
         AppUser readAppUser = app.getApplicationUser(appUser2.getId())
         assertThat readAppUser.getCredentials().getUserName(), equalTo("updated-"+user2.getProfile().getEmail())
     }
-}
\ No newline at end of file
+
+    @Test
+    void csrTest() {
+        Client client = getClient()
+
+        String label = "app-${uniqueTestName}"
+        Application app = client.instantiate(OpenIdConnectApplication)
+            .setLabel(label)
+            .setSettings(client.instantiate(OpenIdConnectApplicationSettings)
+                .setOAuthClient(client.instantiate(OpenIdConnectApplicationSettingsClient)
+                    .setClientUri("https://example.com/client")
+                    .setLogoUri("https://example.com/assets/images/logo-new.png")
+                    .setRedirectUris(["https://example.com/oauth2/callback",
+                                      "myapp://callback"])
+                    .setResponseTypes([OAuthResponseType.TOKEN,
+                                       OAuthResponseType.ID_TOKEN,
+                                       OAuthResponseType.CODE])
+                    .setGrantTypes([OAuthGrantType.IMPLICIT,
+                                    OAuthGrantType.AUTHORIZATION_CODE])
+                    .setApplicationType(OpenIdConnectApplicationType.NATIVE)
+                    .setTosUri("https://example.com/client/tos")
+                    .setPolicyUri("https://example.com/client/policy")))
+            .setCredentials(client.instantiate(OAuthApplicationCredentials)
+                .setOAuthClient(client.instantiate(ApplicationCredentialsOAuthClient)
+                    .setClientId(UUID.randomUUID().toString())
+                    .setAutoKeyRotation(true)
+                    .setTokenEndpointAuthMethod(OAuthEndpointAuthenticationMethod.CLIENT_SECRET_POST)))
+        client.createApplication(app)
+        registerForCleanup(app)
+
+        assertThat(app.getStatus(), equalTo(Application.StatusEnum.ACTIVE))
+
+        // create csr metadata
+        CsrMetadata csrMetadata = client.instantiate(CsrMetadata)
+              .setSubject(client.instantiate(CsrMetadataSubject)
+                  .setCountryName("US")
+                  .setStateOrProvinceName("California")
+                  .setLocalityName("San Francisco")
+                  .setOrganizationName("Okta, Inc.")
+                  .setOrganizationalUnitName("Dev")
+                  .setCommonName("SP Issuer"))
+              .setSubjectAltNames(client.instantiate(CsrMetadataSubjectAltNames)
+                  .setDnsNames(["dev.okta.com"]))
+
+        // generate csr with metadata
+        Csr csr = app.generateCsr(csrMetadata)
```

Comment:
+1 for the case change.  

My pref is that his classname is spelled out into it's full name `CertificateSigningRequest` but I'm not going to block on that (as it isn't my call)